File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/upload_file_with_par_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UploadFileWithParDetails(object):
    """
    The data to create/update a file.
    """

    #: A constant which can be used with the action property of a UploadFileWithParDetails.
    #: This constant has a value of "CREATE"
    ACTION_CREATE = "CREATE"

    #: A constant which can be used with the action property of a UploadFileWithParDetails.
    #: This constant has a value of "REFRESH"
    ACTION_REFRESH = "REFRESH"

    #: A constant which can be used with the action property of a UploadFileWithParDetails.
    #: This constant has a value of "UPDATE"
    ACTION_UPDATE = "UPDATE"

    def __init__(self, **kwargs):
        """
        Initializes a new UploadFileWithParDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param action:
            The value to assign to the action property of this UploadFileWithParDetails.
            Allowed values for this property are: "CREATE", "REFRESH", "UPDATE"
        :type action: str

        :param e_tag:
            The value to assign to the e_tag property of this UploadFileWithParDetails.
        :type e_tag: str

        :param size:
            The value to assign to the size property of this UploadFileWithParDetails.
        :type size: int

        :param system_tags:
            The value to assign to the system_tags property of this UploadFileWithParDetails.
        :type system_tags: dict(str, dict(str, object))

        """
        self.swagger_types = {
            'action': 'str',
            'e_tag': 'str',
            'size': 'int',
            'system_tags': 'dict(str, dict(str, object))'
        }

        self.attribute_map = {
            'action': 'action',
            'e_tag': 'eTag',
            'size': 'size',
            'system_tags': 'systemTags'
        }

        self._action = None
        self._e_tag = None
        self._size = None
        self._system_tags = None

    @property
    def action(self):
        """
        Gets the action of this UploadFileWithParDetails.
        Action type on create or close.

        Allowed values for this property are: "CREATE", "REFRESH", "UPDATE"


        :return: The action of this UploadFileWithParDetails.
        :rtype: str
        """
        return self._action

    @action.setter
    def action(self, action):
        """
        Sets the action of this UploadFileWithParDetails.
        Action type on create or close.


        :param action: The action of this UploadFileWithParDetails.
        :type: str
        """
        allowed_values = ["CREATE", "REFRESH", "UPDATE"]
        if not value_allowed_none_or_none_sentinel(action, allowed_values):
            raise ValueError(
                "Invalid value for `action`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._action = action

    @property
    def e_tag(self):
        """
        Gets the e_tag of this UploadFileWithParDetails.
        Etag that needs to be updated.


        :return: The e_tag of this UploadFileWithParDetails.
        :rtype: str
        """
        return self._e_tag

    @e_tag.setter
    def e_tag(self, e_tag):
        """
        Sets the e_tag of this UploadFileWithParDetails.
        Etag that needs to be updated.


        :param e_tag: The e_tag of this UploadFileWithParDetails.
        :type: str
        """
        self._e_tag = e_tag

    @property
    def size(self):
        """
        Gets the size of this UploadFileWithParDetails.
        Size of the file needed when closed.


        :return: The size of this UploadFileWithParDetails.
        :rtype: int
        """
        return self._size

    @size.setter
    def size(self, size):
        """
        Sets the size of this UploadFileWithParDetails.
        Size of the file needed when closed.


        :param size: The size of this UploadFileWithParDetails.
        :type: int
        """
        self._size = size

    @property
    def system_tags(self):
        """
        Gets the system_tags of this UploadFileWithParDetails.
        System tags for this resource. Each key is predefined and scoped to a namespace.

        Example: `{\"orcl-cloud\": {\"free-tier-retained\": \"true\"}}`


        :return: The system_tags of this UploadFileWithParDetails.
        :rtype: dict(str, dict(str, object))
        """
        return self._system_tags

    @system_tags.setter
    def system_tags(self, system_tags):
        """
        Sets the system_tags of this UploadFileWithParDetails.
        System tags for this resource. Each key is predefined and scoped to a namespace.

        Example: `{\"orcl-cloud\": {\"free-tier-retained\": \"true\"}}`


        :param system_tags: The system_tags of this UploadFileWithParDetails.
        :type: dict(str, dict(str, object))
        """
        self._system_tags = system_tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/UploadFileWithParDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The data to create/update a file.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UploadFileWithParDetails.Builder.class)

public final class UploadFileWithParDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"action", "eTag", "size", "systemTags"})
    public UploadFileWithParDetails(Action action, String eTag, Long size, java.util.Map<String, java.util.Map<String, Object>> systemTags) {
        super();
        this.action = action;
        this.eTag = eTag;
        this.size = size;
        this.systemTags = systemTags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Action type on create or close.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("action")
        private Action action;

                /**
         * Action type on create or close.
         * @param action the value to set
         * @return this builder
         **/
        

        public Builder action(Action action) {
        this.action = action;
        return this;
        }
            /**
     * Etag that needs to be updated.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("eTag")
        private String eTag;

                /**
         * Etag that needs to be updated.
         * @param eTag the value to set
         * @return this builder
         **/
        

        public Builder eTag(String eTag) {
        this.eTag = eTag;
        return this;
        }
            /**
     * Size of the file needed when closed.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("size")
        private Long size;

                /**
         * Size of the file needed when closed.
         * @param size the value to set
         * @return this builder
         **/
        

        public Builder size(Long size) {
        this.size = size;
        return this;
        }
            /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("systemTags")
        private java.util.Map<String, java.util.Map<String, Object>> systemTags;

                /**
         * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
         * @param systemTags the value to set
         * @return this builder
         **/
        

        public Builder systemTags(java.util.Map<String, java.util.Map<String, Object>> systemTags) {
        this.systemTags = systemTags;
        return this;
        }


        public UploadFileWithParDetails build() {
            UploadFileWithParDetails model = new UploadFileWithParDetails(this.action
                    , this.eTag
                    , this.size
                    , this.systemTags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UploadFileWithParDetails model) {
                this.action(model.getAction());
    this.eTag(model.getETag());
    this.size(model.getSize());
    this.systemTags(model.getSystemTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    

    /**
     * Action type on create or close.
     **/
    public enum Action {
        Create("CREATE"),
        Refresh("REFRESH"),
        Update("UPDATE"),
        ;

        

        private final String value;
        private static java.util.Map<String, Action> map;

        static {
            map = new java.util.HashMap<>();
            for (Action v : Action.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Action(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Action create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Action: " + key);
        }
    };
        /**
     * Action type on create or close.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("action")
    private final Action action;

        /**
     * Action type on create or close.
     * @return the value
     **/
    
    public Action getAction() {
        return action;
    }


        /**
     * Etag that needs to be updated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("eTag")
    private final String eTag;

        /**
     * Etag that needs to be updated.
     * @return the value
     **/
    
    public String getETag() {
        return eTag;
    }


        /**
     * Size of the file needed when closed.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("size")
    private final Long size;

        /**
     * Size of the file needed when closed.
     * @return the value
     **/
    
    public Long getSize() {
        return size;
    }


        /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("systemTags")
    private final java.util.Map<String, java.util.Map<String, Object>> systemTags;

        /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     * @return the value
     **/
    
    public java.util.Map<String, java.util.Map<String, Object>> getSystemTags() {
        return systemTags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UploadFileWithParDetails(");
        sb.append("action=").append(String.valueOf(this.action));
        sb.append(", eTag=").append(String.valueOf(this.eTag));
        sb.append(", size=").append(String.valueOf(this.size));
        sb.append(", systemTags=").append(String.valueOf(this.systemTags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UploadFileWithParDetails)) {
            return false;
        }

        UploadFileWithParDetails other = (UploadFileWithParDetails) o;
        return java.util.Objects.equals(this.action, other.action) &&
            java.util.Objects.equals(this.eTag, other.eTag) &&
            java.util.Objects.equals(this.size, other.size) &&
            java.util.Objects.equals(this.systemTags, other.systemTags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.action == null ? 43 : this.action.hashCode());
        result = (result * PRIME) + (this.eTag == null ? 43 : this.eTag.hashCode());
        result = (result * PRIME) + (this.size == null ? 43 : this.size.hashCode());
        result = (result * PRIME) + (this.systemTags == null ? 43 : this.systemTags.hashCode());
        return result;
    }


}
